bukkit/src/main/java/com/denizenscript/depenizen/bukkit/commands/effectlib/EffectLibCommand.java

```diff
@@ -0,0 +1,217 @@
+
+            if (!scriptEntry.hasObject("target")
+                    && arg.matchesPrefix("target")) {
+                scriptEntry.addObject("target", arg.asType(dEntity.class));
+            }
+
+            else if (!scriptEntry.hasObject("duration")
+                    && arg.matchesPrefix("duration")) {
+                scriptEntry.addObject("duration", arg.asType(Duration.class));
+            }
+
+            else if (!scriptEntry.hasObject("location")
+                    && arg.matchesPrefix("location")) {
+                scriptEntry.addObject("location", arg.asElement());
+            }
+
+            else if (!scriptEntry.hasObject("action")
+                    && arg.matchesEnum(Action.values())) {
+                scriptEntry.addObject("action", Action.valueOf(arg.getValue().toUpperCase()));
+            }
+
+            else {
+                arg.reportUnhandled();
+            }
+
+        }
+
+        if (!scriptEntry.hasObject("action")) {
+            throw new InvalidArgumentsException("Effect not specified!");
+        }
+
+        if (!scriptEntry.hasObject("duration")) {
+            throw new InvalidArgumentsException("Duration not specified!");
+        }
+
+        if (!scriptEntry.hasObject("target")) {
+            if (((BukkitScriptEntryData) scriptEntry.entryData).hasPlayer()) {
+                scriptEntry.addObject("target", ((BukkitScriptEntryData) scriptEntry.entryData).getPlayer().getDenizenEntity());
+            }
+        }
+
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void execute(final ScriptEntry scriptEntry) throws CommandExecutionException {
+
+        dEntity target = (dEntity) scriptEntry.getObject("target");
+        Action action = (Action) scriptEntry.getObject("action");
+        Duration duration = (Duration) scriptEntry.getObject("duration");
+        dLocation location = (dLocation) scriptEntry.getObject("location");
+
+        // Report to dB
+        dB.report(scriptEntry, getName(), (target != null ? target.debug() : "")
+                + (action != null ? action.toString() : "")
+                + (duration != null ? duration.debug() : "")
+                + (location != null ? location.debug() : ""));
+
+        if (target == null && location == null) {
+            dB.echoError(scriptEntry.getResidingQueue(), "Target not found!");
+            return;
+        }
+        if (action == null) {
+            dB.echoError(scriptEntry.getResidingQueue(), "Effect type not specified!");
+            return;
+        }
+        if (duration == null) {
+            dB.echoError(scriptEntry.getResidingQueue(), "Duration not specified!");
+            return;
+        }
+
+        int ticks = duration.getTicksAsInt();
+        Plugin plugin = Support.getPlugin(EffectLibSupport.class);
+        EffectManager effectManager = new EffectManager(plugin);
+
+        // TODO: Find a better way to handle all effects and add custom ones as well
```

Comment:
This isn't particularly worth having unless/until there's an automated way to trigger any effect type, as mentioned by this TODO line...

Reply:
There are Configuration-Driven Effects explained on the webpage for EffectLib https://dev.bukkit.org/projects/effectlib
I am not sure how it works, the plugin does not generate any yaml files so I believe this could be done with a new type script if that's possible. Or add something to the config file in Depenizen?

Reply:
probably have a script type for that I guess